### PR TITLE
gateway RemoteGraphQLDataSource: throw GraphQLError, not ApolloError

### PR DESCRIPTION
**Pull request by glasser**
This is part of
https://github.com/apollographql/apollo-server/issues/6057 (which is
itself part of
https://github.com/apollographql/apollo-server/issues/6719). We are
trying to break the dependency of Gateway on Server so that (among other
things) it is easier to have a single version of Gateway that works with
both the current AS3 and the upcoming AS4.

In AS4, we are removing the ApolloError class and its subclasses.
Instead, we will just use GraphQLError directly. See:

https://www.apollographql.com/docs/apollo-server/v4/migration#apolloerror
https://www.apollographql.com/docs/apollo-server/v4/migration#built-in-error-classes
https://github.com/apollographql/apollo-server/pull/6355
https://github.com/apollographql/apollo-server/pull/6705

This commit changes RemoteGraphQLDataSource to throw GraphQLError
instead of ApolloError. The `code` extension will still be the same.
(The `name` field of the thrown Error will no longer be eg
`AuthenticationError`, though; this does not affect the error as
serialized in GraphQL.)

This is technically slightly backwards-incompatible (eg, the method
errorFromResponse is public and now returns GraphQLError instead of the
tighter ApolloError) but this doesn't seem likely to affect many users.
We can adjust based on feedback if necessary.

### CODE REVIEW

**Comment by glasser** [comment]
cc @trevor-scheer

File: gateway-js/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ This CHANGELOG pertains only to Apollo Federation packages in the 2.x range. The
  - Fix fragment reuse in subgraph fetches [PR #1911](https://github.com/apollographql/federation/pull/1911).
  - Allow passing a custom `fetcher` [PR 1997#](https://github.com/apollographql/federation/pull/1997).
   - __UNBREAKING__: Previous 2.1.0 alphas removed the custom fetcher for Apollo Uplink. This re-adds that parameter, and requires the fetcher to have the `AbortSignal` interface https://fetch.spec.whatwg.org/#requestinit.
+- The method `RemoteGraphQLDataSource.errorFromResponse` now returns a `GraphQLError` (as defined by `graphql`) rather than an `ApolloError` (as defined by `apollo-server-errors`). [PR #2028](https://github.com/apollographql/federation/pull/2028)
```
Comment (benweatherman) [inline gateway-js/CHANGELOG.md:8]: **suggestion** Would you mind adding a `__BREAKING__` sub-bullet to this about the change in return type? I agree it's _probably_ low-impact but might as well be explicit.

**Review by benweatherman — APPROVED** [review]
🐐

File: gateway-js/CHANGELOG.md (hunk 1 above)
Comment (glasser) [reply gateway-js/CHANGELOG.md:8]: Done

Comment (glasser) [reply gateway-js/CHANGELOG.md:8]: also tweaked some formatting in some of the lines above

**Pull request merged**